packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js

```diff
@@ -622,4 +622,182 @@ QUnit.module('SmartPaste', () => {
             });
         });
     });
+
```

Comment:
Suggestions for tests:

1. Edge cases for invokeConditionally:

Handling rejected Promise in cancel
Behavior when cancelCallback is undefined

2. Integration tests:

Verify complete flow: smartPaste -> onSmartPasting -> (cancel=false) -> field updates -> onSmartPasted

3. AIResult typing:

Tests for array values (string[]) in aiResult